compiler/src/dotty/tools/dotc/core/Contexts.scala

```diff
@@ -389,6 +389,30 @@ object Contexts {
     def useColors: Boolean =
       base.settings.color.value == "always"
 
+    /** Classloader for loading class accessible from -toolcp */
+    lazy val toolClassloader: ClassLoader = {
+      val urls = ctx.settings.toolcp.value.split(":").map(p => new java.io.File(p).toURI.toURL)
+      new java.net.URLClassLoader(urls, classloader)
+    }
+
+    /** Classloader for loading class accessible from -classpath */
+    lazy val classloader: ClassLoader = {
+      val classpath = ctx.platform.classPath.asURLs.toArray
+      new java.net.URLClassLoader(classpath, getClass.getClassLoader)
+    }
+
+    /** Are macros enabled? */
+    lazy val macrosEnabled: Boolean = {
+      try {
+        Class.forName("scala.meta.eden.package", false, toolClassloader)
+        Class.forName("scala.meta.Tree", false, classloader)
```

Comment:
Normally, we expect a meta block to return meta trees. Of course, we can check for `scala.meta.package$` instead.